### PR TITLE
MdeModulePkg: Enable Data Terminal at end of serial

### PR DESCRIPTION
**Pull request by kenlautner**
# Description

When a Serial device resets, the Modem Control Register Data Terminal Ready and Request to Send need to be cleared also. Otherwise the registers will be left in their previous state, and the connected device will not be able to transmit data.

<_For each item, place an "x" in between `[` and `]` if true. Example: `[x]` (you can also check items in GitHub UI)_>

<_Create the PR as a Draft PR if it is only created to run CI checks._>

<_Delete lines in \<\> tags before creating the PR._>

- [ ] Breaking change?
  - **Breaking change** - Does this PR cause a break in build or boot behavior?
  - Examples: Does it add a new library class or move a module to a different repo.
- [ ] Impacts security?
  - **Security** - Does this PR have a direct security impact?
  - Examples: Crypto algorithm change or buffer overflow fix.
- [ ] Includes tests?
  - **Tests** - Does this PR include any explicit test code?
  - Examples: Unit tests or integration tests.

## How This Was Tested

Tested on a physical device that was not able to receive data after a SerialReset call because the Dtrc and Rts bits were left in their previous states.  With this fix the device not receives data.

## Integration Instructions

N/A

### CODE REVIEW

File: MdeModulePkg/Universal/Console/TerminalDxe/TerminalConIn.c [hunk 1]
```diff
@@ -95,6 +95,10 @@ TerminalConInReset (
       );
   }
 
+  if (!EFI_ERROR (Status)) {
+    Status = TerminalDevice->SerialIo->SetControl (TerminalDevice->SerialIo, EFI_SERIAL_DATA_TERMINAL_READY|EFI_SERIAL_REQUEST_TO_SEND);
```
Comment (mdkinney) [inline MdeModulePkg/Universal/Console/TerminalDxe/TerminalConIn.c:99]: SerialIo->Reset() is called above.  Should that API reset this state so it is common to all consumers of the Serial I/O Protocol?

Comment (mdkinney) [reply MdeModulePkg/Universal/Console/TerminalDxe/TerminalConIn.c:99]: I do see one Serial I/O Protocol implementation preserves the control settings across Reset():

https://github.com/tianocore/edk2/blob/b158dad150bf02879668f72ce306445250838201/MdeModulePkg/Bus/Pci/PciSioSerialDxe/SerialIo.c#L643

The question is if this is the correct behavior for Serial I/O Protocol.  UEFI Spec description of Serial I/O Reset() does not cover this topic.

Comment (apop5) [reply MdeModulePkg/Universal/Console/TerminalDxe/TerminalConIn.c:99]: Hi Mike,

The Serial I/O just says that it resets the hardware, without details of what that means.

In this instance, in the Terminal Console In Devices, when serial I/O is reset, it is with the intention of being able to consume data from the device as a Simple Text In protocol. 

Most serial devices, on reset, will have the DTR and RTS bits default enabled, which signals an attached device that data can now be sent. But the Uefi code does not make it explicit that DTR and RTS should be set.

This PR is handing the case that if a serial device is expected to be used as a ConIn device, that the device is configured to start receiving data.

**Review by apop5 — APPROVED** [review]

File: MdeModulePkg/Universal/Console/TerminalDxe/TerminalConIn.c (hunk 1 above)
Comment (mdkinney) [reply MdeModulePkg/Universal/Console/TerminalDxe/TerminalConIn.c:99]: What about the console output side of TerminalDxe.  Would that have the same issue?

Comment (apop5) [reply MdeModulePkg/Universal/Console/TerminalDxe/TerminalConIn.c:99]: In the hardware we found this on, con out was working no matter the state of dtr/rts.  It's why this is being scoped to only the con in side.

**Comment by lgao4** [comment]
Seemly, there is no negative impact. Right?

**Comment by apop5** [comment]
> Seemly, there is no negative impact. Right?

Correct. This should have no negative impact because it is setting a the bits that, on most hardware, are already set by default.

**Review by lgao4 — APPROVED** [review]

**Pull request merged**